ts_scripts/install_dependencies.py

```diff
@@ -101,7 +101,23 @@ def install_libgit2(self):
         os.system(f"rm -rf libgit2-1.3.0 && rm libgit2-1.3.0.tar.gz")
 
     def install_maven(self):
-        os.system(f"{self.sudo_cmd}apt-get install -y maven")
+        # default maven version that comes with apt install is 3.6 which is incomaptible with jdk17
+        # https://askubuntu.com/questions/1367854/when-compiling-a-project-maven-throws-an-error
+        # therefore, we need to manually install maven version 3.8.6 that is compatible with jdk17
+        MAVEN_VERSION = "3.8.6"
+        MAVEN_CONFIG_FILE = "/etc/profile.d/maven.sh"
+        os.system(
+            f"wget https://apache.org/dist/maven/maven-3/{MAVEN_VERSION}/binaries/apache-maven-{MAVEN_VERSION}-bin.tar.gz -P /tmp"
+        )
+        os.system(f"{self.sudo_cmd}tar xf /tmp/apache-maven-*.tar.gz -C /opt")
+        os.system(f"{self.sudo_cmd}rm /tmp/apache-maven-*-bin.tar.gz")
+        os.system(f"{self.sudo_cmd}ln -s /opt/apache-maven-{MAVEN_VERSION} /opt/maven")
```

Comment:
Do we have to use symbolic links? I dislike them because if someone is developing locally then they we will change the state of their machine without them knowing or being able to opt out of it. For some background check out this https://github.com/pytorch/serve/pull/1254

it may make sense to not force a maven install on users

Reply:
Updated to use maven wrapper which takes care of installing the right maven version - https://maven.apache.org/wrapper/. We just need to replace `mvn` with `mvnw`(unix) | `mvnw.cmd`(windows)